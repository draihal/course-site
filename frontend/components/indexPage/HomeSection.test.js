import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestHomeSection from './HomeSection'


describe('HomeSection', () => {
  it('render correctly HomeSection component', () => {
    const tree = renderer.create(<TestHomeSection
      title={'Тестовый title'}
      short_description={'Тестовое описание'}
    />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render HomeSection correctly with given props', () => {
    const { queryByText } = render(<TestHomeSection
      title={'Тестовый title'}
      short_description={'Тестовое описание'}
    />);
    expect(queryByText('Тестовый title')).toBeTruthy();
    expect(queryByText('Тестовое описание')).toBeTruthy();
  });

});