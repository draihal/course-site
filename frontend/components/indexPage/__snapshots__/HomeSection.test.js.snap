// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeSection render correctly HomeSection component 1`] = `
<div
  id="home"
>
  <div
    className="landing"
  >
    <div
      className="home-wrap"
    >
      <div
        className="home-inner"
      />
    </div>
  </div>
  <div
    className="caption text-center"
  >
    <div
      className="container"
    >
      <h1>
        Тестовый title
      </h1>
      <h3>
        Тестовое описание
      </h3>
      <a
        className="btn btn-outline-light btn-lg"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Выбрать
      </a>
    </div>
  </div>
</div>
`;
